src/connectedk8s/azext_connectedk8s/_params.py

```diff
@@ -5,10 +5,13 @@
 # pylint: disable=line-too-long
 
 from argcomplete.completers import FilesCompleter
-from azure.cli.core.commands.parameters import get_location_type
+from azure.cli.core.commands.parameters import get_location_type, get_enum_type
 from azure.cli.core.commands.parameters import (file_type)
 from azure.cli.core.commands.validators import get_default_location_from_resource_group
 
+distribution_types = ["auto", "generic", "openshift", "rancher_RKE", "kind", "GKE", "EKS", "AKS", "AKS-HCI", "CAPZ", "AKS-engine", "TKG"]
+infrastructure_types = ["auto", "generic", "azure", "AWS", "GCP", "Azure_Stack_HC", "Azure_Stack_Hub", "Azure_Stack_Edge", "vSphere"]
```

Comment:
Use following values:

- generic
- AWS
- GCP
- Azure
- AzureStackEdge
- AzureStackHub
- AzureStackHCI
- vSphere

Reply:
One doubt, so if user is providing these values, he will have to type whole "Amazon_Web_Services". Isn't it easy for user for type just --infrastructure AWS?. We can always add a logic wherever we want to display(like in portal) to transform from AWS to Amazon Web Services.